std/markdown/markdown_native.go

```diff
@@ -33,3 +36,51 @@ func convertString(source string) string {
 	}
 	return buf.String()
 }
+
+func getKeywordFlag(opts Map, name string, def bool) bool {
+	ok, entry := opts.Get(MakeKeyword(name))
+	if !ok {
+		return def
+	}
+	val, ok := entry.(Boolean)
+	if !ok {
+		panic(RT.NewError("flags must be a boolean"))
+	}
+	return val.B
+}
+
+func convertStringOpts(source string, opts Object) string {
+	options, ok := opts.(Map)
+	if !ok {
+		panic(RT.NewError("Options must be a map"))
+	}
+
+	renderOptions := []renderer.Option{}
+	if flag := getKeywordFlag(options, "with-hard-wraps?", true); flag {
+		renderOptions = append(renderOptions, html.WithHardWraps())
+	}
+	if flag := getKeywordFlag(options, "with-xhtml?", true); flag {
+		renderOptions = append(renderOptions, html.WithXHTML())
+	}
+	if flag := getKeywordFlag(options, "with-unsafe?", true); flag {
+		renderOptions = append(renderOptions, html.WithUnsafe())
+	}
+	md := goldmark.New(
```

Comment:
This is largely a duplication of `convertString`. Can you refactor the code to avoid duplication?

Reply:
When I copied the whole method I definitely told myself I would need to go back and fix the duplication. I guess I just got a little excited 😬